### ISSUE TITLE
RedundantCondition with falsable-return functions

### ISSUE THREAD

**Issue by weirdan** [vimeo/psalm#825]
Psalm has started complaining about excessive checks on function results for functions that technically can return false:
```php
<?php
  $format = random_bytes(10);
  $dt = new DateTime;
  $formatted = $dt->format($format);
  assert(false !== $formatted);
```
Here, the assertion is not redundant (and I'd argue any formal assertions (using `assert()`) should not be complained about, to support things like DbC).

**Comment by weirdan** [comment]
Not exactly. I'd like to keep those false checks optional, but allowed. Not required, nor forbidden. That's ignore-falsable-return I suppose (which I was likely unaware of when I posted that issue).

Here, that assert was added to shut phpstan up, and in other places we may use similar asserts to improve autocompletion.

**Comment by muglug** [comment]
Ok, I've given CallMap methods the same treatment as CallMap functions. Are there any builtin methods where we shouldn't ignore possibly `false` issues?

**Comment by weirdan** [comment]
I'm pretty sure there will be. Off the top of my head not checking the return value of `Memcache::get()` for `=== false` is almost always a problem.
